core/src/main/scala/kafka/admin/BrokerVersionCommand.scala

```diff
@@ -0,0 +1,84 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kafka.admin
+
+import java.util.Properties
+
+import kafka.common.Config
+import kafka.utils.CommandLineUtils
+import org.apache.kafka.common.utils.Utils
+import org.apache.kafka.clients.CommonClientConfigs
+import joptsimple._
+
+import scala.util.{Failure, Success}
+
+
+/**
+ * A command for retrieving broker version information.
+ */
+object BrokerVersionCommand extends Config {
+
+  def main(args: Array[String]): Unit = {
+    val opts = new BrokerVersionCommandOptions(args)
+    val adminClient = createAdminClient(opts)
+    val brokerMap = adminClient.listAllBrokerVersionInfo()
+    brokerMap.foreach { case (broker, versionInfoOrError) =>
+      versionInfoOrError match {
+        case Success(v) => print(s"${broker.toString} -> ${v.toString(true)}\n")
+        case Failure(v) => print(s"${broker.toString} -> ${v}\n")
```

Comment:
Hmm, not sure what you mean by "the no-args versions are not needed."  The no-args version of NodeVersionInfo provides the single-line output, which you would want in log files (for example).  The one-arg version, which we are using here, allows us to supply a boolean which gives a multi-line version which is a little more readable, but more verbose, as suggested by @apurvam.

I'll add an ERROR prefix to the Failure line.